### PR TITLE
base1: Modernize cockpit.file().read()

### PR DESCRIPTION
**Pull request by mvollmer**
So that we can write

    const [content, tag] = await file.read2()

### CODE REVIEW

**Comment by mvollmer** [comment]
I think it's fine to use

    const content = file.read()

We don't need to deprecate it.

File: pkg/lib/cockpit.js [hunk 1]
```diff
@@ -2243,6 +2244,10 @@ function factory() {
             return read_promise;
         }
 
+        function read2() {
+            return read().then((content, tag) => [content, tag]);
```
Comment (cockpituous) [inline pkg/lib/cockpit.js:2248]: These 2 added lines are not executed by any test.

**Comment by martinpitt** [comment]
I'm a bit torn here. It's kind of nice to write it in that style, but I find foo2() style functions rather poorly named. At the same  time it's hard to find a succinct and descriptive name for it. Also, it's actually rare that we need the tag, so for these rare cases writing promise-style isn't perhaps too bad?

You can also use async with a little crutch:

```
> const [content, tag] = await cockpit.file("/etc/issue").read().then((c,t) => [c,t])
Array [ "\\S\nKernel \\r on an \\m (\\l)\n\n", "1:1050749-0.0-100644-0-0" ]
```

(`arguments` doesn't work for arrows)

If you really like/want this, no big veto, but of course it should be documented properly then. And this should replace at least one user in our pages.

**Comment by mvollmer** [comment]
> so for these rare cases writing promise-style isn't perhaps too bad?

No, it isn't, but don't we want to get away from multi-value Promises? Can TypeScript even type them correctly? I think the current type of `modify` is wrong, for example (and we might get away with that because no code uses the result of modify, maybe). I really need to start getting my hands dirty with tsc...

**Comment by mvollmer** [comment]
> but I find foo2() style functions rather poorly named.

Yeah, naming is hard... :-) Just look at wait3 and wait4 in libc... In this case the "2" actually has meaning, kind of: the function returns two values. But read_with_tag would work as well, of course.

**Comment by mvollmer** [comment]
> If you really like/want this

Personally, I don't want this. I made this PR to illustrate what I was saying yesterday on IRC.

**Comment by martinpitt** [comment]
> don't we want to get away from multi-value Promises?

If we'd do it all over again, for sure. We'd return an object with `content`, `tag`, and possibly other meta-information.

We could mark .read() as deprecated and move everything to a new  .read_with_tag() or .load() or so. 

> Can TypeScript even type them correctly?

I don't know :cry: I just checked cockpit-files, and we only use cockpit.file.replace() there, not .modify().

**Comment by mvollmer** [comment]
This was just FYI, let's remove it from my ever growing list of unfinished PRs... :-)

**Pull request closed without merging**